### ISSUE TITLE
feature: Add keymap for `:Telescope quickfixhistory`

### ISSUE THREAD

**Issue by osamuaoki** [LazyVim/LazyVim#3813]
### Did you check the docs?

- [X] I have read all the LazyVim docs
- [X] This is not a request for a new or existing extra (see above)

### Is your feature request related to a problem? Please describe.

Although we can perform down-selection of quickfix list with `<C-q>` or `<M-q>`, currently it is a bit awkward to manage quickfix history generated by such actions.

This is because, LazyVim doesn't define keymap for existing `:Telescope quickfixhistory` while `:Telescope quickfix` and `:Telescope loclist` are already assigned to `<leader>sq` and `<leader>sl`.

For consistency and ease of use, it will be nice to set all Quickfix related key maps under `<leader>x`  as:
* `:Telescope quickfixhistory` (add: `<leader>xh`),
* `:Telescope quickfix` (change: `<leader>xt`)
* `:Telescope loclist` (change: `<leader>xT`)
 
Note:  `<leader>xq`, `<leader>xQ`, `<leader>xl`, `<leader>xL` are already taken
* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * 

### Describe the solution you'd like

I have following in my local setting.  Making these as a part of LazyVim via its `lua/keymaps.lua` may be good idea.
```lua
-- Telescope quickfix / location list can perform down-selection.
-- <tab> / <S-Tab> to mark
-- <M-q> / <C-q> to down-select list to quickfix (marked / unmarked)
map("n", "<leader>xt", "<cmd>Telescope quickfix<cr>", { desc = "Quickfix List (Telescope)" }) -- same as <leader>sq
map("n", "<leader>xT", "<cmd>Telescope loclist<cr>", { desc = "Location List (Telescope)" })   -- same as <leader>sl
map("n", "<leader>xh", "<cmd>Telescope quickfixhistory<cr>", { desc = "Quickfix History (Telescope)" })
map("n", "<leader>xc", vim.cmd.cclose,{ desc = "Close Quickfix List" })
map("n", "<leader>xC", vim.cmd.lclose, { desc = "Close Location List" })
```

### Describe alternatives you've considered

I also thought about changing `<leader>xq` and `<leader>xl` as toggling one to avoid creating separate `:cclose` / `:lclose` bindings. 

### Additional context

Use of noice.nvim is making it more difficult to use `:message` to check `:chistory`.  Adding above makes this non-issue.

This feature also allow us to avoid using `:colder` / `:cnewer` which are error prone.

**Comment by dpetka2001** [comment]
I believe the mappings for Telescope should go under `keys` in Telescope spec, because there is also `fzf-lua` alternative (that quite a few users use), in order for the mappings to take effect only if Telescope is enabled.

**Comment by osamuaoki** [comment]
I see and I understand.  What do you suggest as good choices?

**Comment by osamuaoki** [comment]
I send PR with the narrow scope as suggested.

Come to think of, as for #190, I was actually using `ZZ` which also worked.  So no need.

**Comment by dpetka2001** [comment]
I believe in `fzf-lua` the corresponding command is `FzfLua quickfix_stack`, but I don't use it that much. Just leave it be. It's just one keymap and users can add it to their personal configuration if they really want to. Same goes for the Telescope mapping in my personal opinion as just a simple user, but the maintainer has the final say.

**Comment by osamuaoki** [comment]
Yah, I found up myself, too.  It uses similar but updated internal  call. (I updated PR and switched to use fzf-lua here.)

* If one use quickfix primarily for error fix jump only, maybe quickfix history/stack is not so useful.
* If one use it with `:vim ...` with `:cdo` and uses its existing down-selection UI, this history/stack management is very handy.

Considering these are upstream documented UI functionality of both telescope and fzf-lua in their README, it is not a bad idea to support them.  But like you said, its up to the maintainer. 

(We can always add them to our keymaps.lua)
